### DISCUSSION TITLE
website content not loading on github pages

### DISCUSSION THREAD

**Discussion by Kazzzzzyyyy** [community/community#81338] [New to GitHub]
this is my code for a linear gradient background. 

<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <style>
        body {
            height: 100vh;
            margin: 0;
            background: rgb(67, 127, 172);
            background: linear-gradient(90deg, rgba(67, 127, 172, 1) 5%, rgba(27, 82, 136, 1) 96%);
        }
    </style>
</head>
<body>
</body>
</html>

Unfortunately, the github page shows only the name of the repository. The browser console also reports no errors. Any suggestions?

**Comment by faridnec** [comment]
Hey, can you make sure your CSS code is within a `.css` file, and linked correctly to your HTML file?
If using style tags within the HTML, verify they're placed within the `<head>` section. If it does not help, confirm that your GitHub Pages site is published from the correct branch (usually `main` or `master`).

provide more details if it still does not work

**Reply by Sandeepkasturi** [reply]
That true Buddy, But we can actually write CSS in any part of the HTML code within the <html> tags. Although its sort of solution for that problem. :"Website content not loading on github pages":.

If you are good at Python use streamlit for that.

**Comment by varunpats** [comment]
Hey, it's a good practice to create a separate file for CSS. I would suggest you to do the same.

Moreover, I tried replicating the same and seems like it is working, please check [this](https://onecompiler.com/html/3zx6uayuu). That means there might be some issue while publishing site on github pages.

Hope this helps.

**Comment by Sandeepkasturi** [comment]
Try to use External CSS, Use proper tags and check for console errors by inspecting the code.
It seems like you are encountering an issue with the display of your GitHub Pages, and you've provided the code for a linear gradient background in the `<style>` element. There could be several reasons why the content is not loading as expected. Here are some suggestions to troubleshoot the issue:

1. **Check File Paths:**
   Ensure that the paths to your HTML, CSS, and other relevant files are correct. GitHub Pages relies on the correct file structure. Verify that your HTML file is in the root or specified directory, and the CSS file is linked properly.

2. **Confirm GitHub Pages Settings:**
   Make sure GitHub Pages is enabled for your repository and is set to the correct branch. You can check this in the repository settings under the "Pages" section.

3. **Console Errors:**
   Even though you mentioned no errors in the console, recheck for any warnings or errors that might not be immediately visible. Open the browser console (usually by pressing `F12` or right-clicking and selecting "Inspect," then going to the "Console" tab) and look for any relevant messages.

4. **GitHub Pages Build Process:**
   GitHub Pages might take a few moments to build and deploy your site. Check the GitHub Actions or repository settings to ensure that the deployment process is completing successfully.

5. **Jekyll Configuration (if applicable):**
   If you are using Jekyll for your GitHub Pages site, make sure your Jekyll configuration is correct. Check for any Jekyll-related errors or warnings in the build process.

6. **CSS Specificity:**
   Ensure that there are no conflicting styles elsewhere in your CSS that might be overriding the styles you've defined for the body. Check for any inline styles or styles from external libraries.

7. **Browser Cache:**
   Clear your browser cache or try accessing the GitHub Pages site in an incognito/private browsing window to rule out any caching issues.

8. **Syntax Errors:**
   Double-check your CSS code for any syntax errors. Although you mentioned no errors in the console, it's always good to ensure there are no issues with the CSS code itself.

If the issue persists after checking these points, providing more information about your project structure or a link to the repository would be helpful for further assistance.
Tell me further if you have any doubts.

**Comment by seyLu** [comment]
Hi @Kazzzzzyyyy,

GitHub Pages looks for `index.html` at the root directory of your repo. If it can't find it, then it will look for `index.md` or `README.md`.

![image](https://github.com/community/community/assets/98249191/096a2ace-a4d1-41fc-8129-4941a91c5c6a)

![image](https://github.com/community/community/assets/98249191/df3e8510-6120-45ed-baac-c82a0aff000c)

In fact, if we go to `Code.html`, we would see:

![image](https://github.com/community/community/assets/98249191/0ee9b1e8-8ab8-46a2-af66-3189d071b5f8)

You can read more at [GitHub Pages docs](https://docs.github.com/en/pages/getting-started-with-github-pages/creating-a-github-pages-site#creating-your-site).

Hope that helps and Happy Holidays! :)
